src/in-view.js

```diff
@@ -17,40 +17,46 @@ const inView = () => {
     * each registry.
     */
     const interval = 100;
-    const triggers = ['scroll', 'resize', 'load'];
+    const triggers = ['scroll', 'resize'];
 
     /**
     * Maintain a hashmap of all registries, a history
     * of selectors to enumerate, and an options object.
     */
     let selectors = { history: [] };
-    let options   = { offset: {}, threshold: 0, test: inViewport };
+    let options   = { offset: {}, threshold: 0, test: inViewport, async: true };
 
     /**
     * Check each registry from selector history,
     * throttled to interval.
     */
     const check = throttle(() => {
         selectors.history.forEach(selector => {
-            selectors[selector].check();
+            const registry = selectors[selector];
+            if (options.async) {
+                let elements = [].slice.call(document.querySelectorAll(selector));
+                if (elements.length !== registry.elements.length) {
+                    registry.elements = elements;
+                }
+            }
+            registry.check();
         });
     }, interval);
 
     /**
     * For each trigger event on window, add a listener
     * which checks each registry.
     */
-    triggers.forEach(event =>
-        addEventListener(event, check));
+    triggers.forEach(event => window.addEventListener(event, check));
 
     /**
     * If supported, use MutationObserver to watch the
     * DOM and run checks on mutation.
     */
     if (window.MutationObserver) {
-        addEventListener('DOMContentLoaded', () => {
+        isReady(() => {
             new MutationObserver(check)
-                .observe(document.body, { attributes: true, childList: true, subtree: true });
+                .observe(document, { attributes: true, childList: true, subtree: true });
```

Comment:
It depends on the time when the library is loaded and used. If the DOM is not ready upfront as creating a browser extension its better to specify it as general as possible.